ci/buildkite-pipeline.sh

```diff
@@ -14,8 +14,12 @@ if [[ -n $CI_PULL_REQUEST ]]; then
   pr_number=${BASH_REMATCH[1]}
   echo "get affected files from PR: $pr_number"
 
+  # Fetch the number of commits in the PR
+  commits_no=$(gh pr view "$pr_number" --json commits --jq '.commits | length')
```

Comment:
Would like to avoid the GitHub API here, instead we could use `git merge-base` to find the LCA.

Reply:
We can use `git merge-base` with branches or commits, but we don't know from buildkite which branch the PR intends to merge to. Likewise, we don't know which commit to compare.